producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/controller/ExecutionEventsController.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.producer.rest.controller
+
+import io.swagger.annotations.{Api, ApiOperation, ApiResponse, ApiResponses}
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Controller
+import org.springframework.web.bind.annotation._
+import za.co.absa.spline.producer.rest.model.ExecutionEvent
+import za.co.absa.spline.producer.service.repo.ExecutionProducerRepository
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Controller
+@Api(tags = Array("execution"))
+class ExecutionEventsController @Autowired()(
+  val repo: ExecutionProducerRepository) {
+
+  import ExecutionContext.Implicits.global
+
+  @PostMapping(Array("/execution-events"))
+  @ApiOperation(
+    value = "Save Execution Events",
+    notes =
+      """
+        Saves a list of Execution Events.
+
+        RequestBody format :
+
+        Array[ExecutionEvent]
+        [
+          // Reference to the Executionplan Id that was triggered
+          planId: UUID
+          // Timestamp of the execution
```

Comment:
is this the start/end or something else?

Reply:
That's actually a million $ question. Actually it's both and neither. While I was writing a detailed explanation I started doubting in what I was going to write. I'll think about it :)

Reply:
Ok, the answer is this -  the `timestamp` is the time of _end_ of write event. So far we were not interested in _start_ time mainly because of the assumption that data batches are processed in isolation. We assume that every job is atomic (in a sense that the state of the data sources read is not change during the job execution). Hence from the lineage perspective (how the data sources are linked, what affects what etc) the job duration is irrelevant.

But it's worth noting that although such assumption makes sense to me, it potentially might not hold for 100% Spline use cases. It also good to track job duration for the monitoring and stats collection purposes. so I've files another ticket for that - #477

Reply:
That sounds good. Could we add a bit of the explanation to the docs?

Reply:
Sure, we'll elaborate on it more in #437. For now I just state that it's the "end" time